source/language/openpulse.rst

```diff
@@ -0,0 +1,609 @@
+  }
+
+Capture Instruction
+-------------------
+
+Acquisition is scheduled by a ``capture`` instruction. This is a special
+``kernel`` function which is specified by a hardware vendor. The measurement
+process is difficult to describe generically due to the wide variety of
+hardware and measurement methods. Like the play instruction, these instructions
+may only appear inside a ``defcal`` block!
+
+The only required parameters are the ``channel`` and the ``frame``.
+
+The following are possible parameters that might be included:
+
+- A "duration" of type ``length``, if it cannot be inferred from other parameters.
+- A "filter" of type ``waveform``, which is dot product-ed with the measured IQ to distill the
+  result into a single IQ value
+
+Again it is up to the hardware vendor to determine the parameters and write a
+kernel definition at the top-level, such as:
+
+.. code-block:: javascript
+
+   // Minimum requirement
+   kernel capture(channel chan, frame output) -> complex[32];
+
+   // A capture command that returns an iq value
+   kernel capture(channel chan, waveform filter, frame output) -> complex[32];
+
+   // A capture command that returns a discrimnated bit
+   kernel capture(channel chan, waveform filter, frame output) -> bit;
+
+   // A capture command that returns a raw waveform data
+   kernel capture(channel chan, length len, frame output) -> waveform;
+
+The return type of a ``capture`` command varies. It could be a raw trace, ie. a
+list of samples taken over a short period of time. It could be some averaged IQ
+value. It could be a classified bit. Or it could even have no return value,
+pushing the results into some buffer which is then accessed outside the program.
+
+For example, the ``capture`` instruction could return raw waveform data that is then
+discriminated using user-defined boxcar and discrimnation ``kernel``s.
+
+.. code-block:: javascript
+
+    // Use a boxcar function to generate IQ data from raw waveform
+    kernel boxcar(waveform input) -> complex[64];
+    // Use a linear discriminator to generate bits from IQ data
+    kernel discriminate(complex[64] iq) -> bit;
+
+    defcal measure $0 -> bit {
+        // Define the channels
+        txchannel m0 = txch($0, "measure");
+        rxchannel cap0 = rxch($0, "capture");
+
+        // Force time of carrier to 0 for consistent phase for discrimination.
+        frame stimulus_frame = newframe(5e9, 0);
+        frame capture_frame = newframe(5e9, 0);
+
+        // Apply measurement stimulus
+        waveform meas_wf = gaussian_square(1.0, 16000dt, 262dt, 13952dt);
+
+        // Play the stimulus
+        play(m0, meas_wf, stimulus_frame);
+        // Align measure and capture channels
+        barrier(stimulus_frame, capture_frame);
+        // Capture transmitted data after interaction with measurement resonator
+        // kernel capture(channel chan, frame capture_frame) -> waveform;
+        waveform raw_output = receive(cap0, meas_pulse.duration);
```

Comment:
What is `receive`? Believe this is lurking from past drafts. Also do we require the `duration` for the `capture`? How else do we know when to stop collecting data?

Reply:
Yeah this is a stale example. I think it should read more like
```
// extern capture(channel chan, length len, frame output) -> waveform;
waveform raw_output = capture(cap0, meas_pulse.duration, capture_frame)
```

Reply:
Fixed in fa97dfa.